lib/modules/platform/gitea/utils.ts

```diff
@@ -7,3 +14,50 @@ export function smartLinks(body: string): string {
 export function trimTrailingApiPath(url: string): string {
   return url?.replace(regEx(/api\/v1\/?$/g), '');
 }
+
+export function getRepoUrl(
+  repo: Repo,
+  gitUrl: GitUrlOption | undefined,
+  endpoint: string
+): string {
+  if (gitUrl === 'ssh') {
+    if (!repo.ssh_url) {
+      throw new Error(CONFIG_GIT_URL_UNAVAILABLE);
+    }
+    logger.debug({ url: repo.ssh_url }, `using SSH URL`);
+    return repo.ssh_url;
+  }
+
+  // Find options for current host and determine Git endpoint
+  const opts = hostRules.find({
+    hostType: PlatformId.Gitea,
+    url: endpoint,
+  });
+
+  if (gitUrl === 'endpoint') {
+    const url = parseUrl(endpoint);
+    if (!url) {
+      throw new Error(CONFIG_GIT_URL_UNAVAILABLE);
+    }
+    url.protocol = url.protocol?.slice(0, -1) ?? 'https';
+    url.username = opts.token ?? '';
+    url.pathname = url.pathname + repo.full_name + '.git';
```

Comment:
```suggestion
    url.pathname = `${url.pathname}${repo.full_name}.git`;
```

prefer string templates